src/main/java/org/iac2/common/utility/Utils.java

```diff
@@ -28,4 +31,38 @@ public static DockerClient createDockerClient(String dockerEngineUrl) {
 
         return DockerClientImpl.getInstance(dockerConfig, httpClient);
     }
+
+    private static URI parseUrl(String value) {
+        try {
+            return new URI(value);
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    private static boolean isIp(String value) {
+        if (value == null) {
+            return false;
+        }
+
+        String IPV4_PATTERN = "(([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\\.){3}" +
```

Comment:
does it have to be an IP here? or are you only trying to distinguish between IP and URLs ?

Reply:
this is intended to match a URL that only has an ip, i.e., no `http(s)://`
IPs with `http(s)://` (or whatever other scheme) captured as a regular URLs